src/System.Windows.Extensions/src/System/Media/SoundPlayer.cs

```diff
@@ -0,0 +1,738 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Runtime.InteropServices;
+using System.Runtime.Serialization;
+using System.Threading;
+
+namespace System.Media
+{
+    [Serializable]
+    [ToolboxItem(false)]
+    public class SoundPlayer : Component, ISerializable
+    {
+        private const int BlockSize = 1024;
+        private const int DefaultLoadTimeout = 10000; // 10 secs
+
+        private Uri _uri = null;
+        private string _soundLocation = string.Empty;
+        private int _loadTimeout = DefaultLoadTimeout;
+
+        // used to lock all synchronous calls to the SoundPlayer object
+        private ManualResetEvent _semaphore = new ManualResetEvent(true);
+
+        // the worker copyThread
+        // we start the worker copyThread ONLY from entry points in the SoundPlayer API
+        // we also set the tread to null only from the entry points in the SoundPlayer API
+        private Thread _copyThread = null;
+
+        // local buffer information
+        private int _currentPos = 0;
+        private Stream _stream = null;
+        private Exception _lastLoadException = null;
+        private bool _doesLoadAppearSynchronous = false;
+        private byte[] _streamData = null;
+        private AsyncOperation _asyncOperation = null;
+        private readonly SendOrPostCallback _loadAsyncOperationCompleted;
+
+        // event
+        private static readonly object s_eventLoadCompleted = new object();
+        private static readonly object s_eventSoundLocationChanged = new object();
+        private static readonly object s_eventStreamChanged = new object();
+
+        public SoundPlayer()
+        {
+            _loadAsyncOperationCompleted = new SendOrPostCallback(LoadAsyncOperationCompleted);
+        }
+
+        public SoundPlayer(string soundLocation) : this()
+        {
+            SetupSoundLocation(soundLocation ?? string.Empty);
+        }
+
+        public SoundPlayer(Stream stream) : this()
+        {
+            _stream = stream;
+        }
+
+        protected SoundPlayer(SerializationInfo serializationInfo, StreamingContext context)
```

Comment:
Similarly, @ViktorHofer, presumably we should delete all of this serialization stuff?